src/qofonosimwatcher.cpp

```diff
@@ -58,8 +57,12 @@ void QOfonoSimWatcher::Private::onOfonoAvailableChanged()
         allSims.clear();
         if (!presentSims.isEmpty()) {
             presentSims.clear();
-            Q_EMIT watcher->presentSimListChanged();
-            Q_EMIT watcher->presentSimCountChanged();
+            if (watcher) {
+                Q_EMIT watcher->presentSimListChanged();
+            }
+            if (watcher) {
```

Comment:
And you also don't know what the signal issuer is going to do after control returns back to it, i.e. whether it needs to access its `this` pointer after that or not. Also, signals are often chained, i.e. one signal is causing another signal which are emitting other signals and in general there's no mechanism to determine whether the object you are about to delete is the sender of one of those signals. One solution is to always use `deleteLater()` for deleting any `QObject` that has any signals, unless it's internal and you know exactly how it's being used and you can guarantee that at this particular moment you are not handling its signal. Or just live with it and fix one crash at a time.